modules/apps/redirect/redirect-web/src/main/java/com/liferay/redirect/web/internal/portlet/action/EditRedirectPatternsMVCActionCommand.java

```diff
@@ -110,10 +114,19 @@ private List<RedirectPatternEntry> _getRedirectPatternEntries(
 			if ((patternString != null) && (destinationURL != null) &&
 				(userAgent != null)) {
 
-				redirectPatternEntries.add(
-					new RedirectPatternEntry(
-						Pattern.compile(patternString), destinationURL,
-						userAgent));
+				try {
+					redirectPatternEntries.add(
+						new RedirectPatternEntry(
+							Pattern.compile(patternString), destinationURL,
+							userAgent));
+				}
+				catch (PatternSyntaxException patternSyntaxException) {
+					_log.error(patternSyntaxException);
+
+					throw new InvalidRedirectionPatternException(
+						patternString + " must contain regular expression",
+						patternSyntaxException);
```

Comment:
I would log a more meaningful message like: 
```java
"The pattern '" + patternString + "' is not a valid regular expression"
```

Otherwise the message we can receive is: ") must contain regular expression"

Reply:
As my previous comment, I wanted to follow the same pattern as the previous behavior

Reply:
"Following the same pattern as before" doesn't always mean "it's fine". IMO we should improve where is possible.